src/main/java/seedu/sudohr/logic/commands/eventCommands/UpdateEventCommand.java

```diff
@@ -0,0 +1,155 @@
+import seedu.sudohr.logic.commands.Command;
+import seedu.sudohr.logic.commands.CommandResult;
+import seedu.sudohr.logic.commands.exceptions.CommandException;
+import seedu.sudohr.model.Model;
+import seedu.sudohr.model.events.Event;
+import seedu.sudohr.model.events.Title;
+import seedu.sudohr.model.person.Person;
+
+/**
+ * Updates the details of an existing event inside SudoHR.
+ */
+public class UpdateEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "updateEvent";
+
+    public static final String MESSAGE_EDIT_EVENT_SUCCESS = "Edited Event: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_EVENT = "This event already exists in the sudohr book.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the event identified "
+            + "by the index number used in the displayed event list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_EVENT_TITLE + "NAME]\n "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_EVENT_TITLE + "edited Event";
+
+    private final Index index;
+    private final EditEventDescriptor editEventDescriptor;
+
+    /**
+     * Creates an UpdateEventCommand to edit the event at the specified {@code index}
+     *
+     * @param index of the person in the filtered person list to edit
+     * @param editPersonDescriptor details to edit the person with
+     */
+
+    public UpdateEventCommand(Index index, EditEventDescriptor editEventDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editEventDescriptor);
+
+        this.index = index;
+        this.editEventDescriptor = editEventDescriptor;
+    }
+
+    /**
+     * Creates and returns a {@code event} with the details of {@code eventToEdit}
+     * edited with {@code editEventDescriptor}.
+     */
+    private static Event createEditedEvent(Event eventToEdit, EditEventDescriptor editEventDescriptor) {
+        assert eventToEdit != null;
+        Title updatedTitle = editEventDescriptor.getTitle().orElse(eventToEdit.getTitle());
+        Set<Person> personsInEvent = eventToEdit.getAttendees();
+
+        return new Event(updatedTitle, personsInEvent);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Event> lastShownList = model.getFilteredEventList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+        }
+
+        Event eventToEdit = lastShownList.get(index.getZeroBased());
+        Event editedEvent = createEditedEvent(eventToEdit, editEventDescriptor);
+
+        if (!eventToEdit.isSameEvent(editedEvent) && model.hasEvent(editedEvent)) {
```

Comment:
correct me if I'm wrong, but isn't it "eventToEdit.isSameEvent(editedEvent) || model.hasEvent(editedEvent)" to check for duplicate?